### DISCUSSION TITLE
User Input

### DISCUSSION THREAD

**Discussion by wawesomeNOGUI** [hajimehoshi/ebiten#1492] [Q&A] [answered]
Hi! As far as I could tell, Go is not able to get keystrokes and other input in real-time and can only get input through STDIN. Can someone explain to me how user input is gathered in ebiten?

**Comment by hajimehoshi** [comment, marked as answer]
Hi,

Are you asking about the implementation? This depends on OSes. Please take a look at the implementation of [GLFW](https://github.com/glfw/glfw).

**Reply by wawesomeNOGUI** [reply]
So GLFW is used to get user input?

**Reply by hajimehoshi** [reply]
Yes

**Reply by wawesomeNOGUI** [reply]
Awesome thanks for the info! Love the ebiten project!
